apps/jubmoji-quest/src/components/GoogleWalletButton.tsx

```diff
@@ -2,79 +2,82 @@ import { loadBackupState, saveBackupState } from "../lib/localStorage";
+
+      // prep an update link if user already has a backup
+      // need to also generate save link in case user didn't actually backup
       if (backup !== undefined && backup.type === "google") {
-        // just updating backup, use Wallet API
-        setUpdateBackup(true);
-        setSaveUrl(
-          `/api/updateGooglePass?number=${jubmojis.length}&serial=${
-            backup.serialNum
-          }&collection=${serializeJubmojiList(jubmojis)}`
+        setUpdateUrl(
+          `/api/updateGooglePass?number=${jubmojis.length}&serial=${backup.serialNum}&collection=${succinctSerialization}`
         );
-        setSerial(backup.serialNum);
-      } else {
-        // first time saving backup, use save URL
-        setUpdateBackup(false);
-
-        // create save URL for wallet
-        const claims = await getClaims({
-          jubmojiWalletBackup: serializeJubmojiList(jubmojis),
-          number: jubmojis.length,
-        });
-        if (!claims) return;
-        const PRIVATE_KEY = await jose.importPKCS8(
-          GOOGLE_WALLET.PRIVATE_KEY,
-          "RS256"
-        );
-        const token = await new jose.SignJWT(claims)
-          .setProtectedHeader({ alg: "RS256", typ: "JWT" })
-          .setIssuedAt() // Add issued at time (iat)
-          .sign(PRIVATE_KEY);
-        setSaveUrl(`https://pay.google.com/gp/v/save/${token}`);
-        setSerial(GOOGLE_WALLET.OBJECT_ID);
       }
+
+      const backupSerial = crypto.randomUUID();
+      const claims = await getClaims({
+        jubmojiWalletBackup: succinctSerializeJubmojiList(jubmojis),
+        number: jubmojis.length,
+        serial: backupSerial,
+        emojis: jubmojis
+          .map((jubmoji) => cardPubKeys[jubmoji.pubKeyIndex].emoji)
+          .join(""),
+      });
+      if (!claims) return;
+      const PRIVATE_KEY = await jose.importPKCS8(
+        GOOGLE_WALLET.PRIVATE_KEY,
+        "RS256"
+      );
+      const token = await new jose.SignJWT(claims)
+        .setProtectedHeader({ alg: "RS256", typ: "JWT" })
+        .setIssuedAt() // Add issued at time (iat)
+        .sign(PRIVATE_KEY);
+      setSaveUrl(`https://pay.google.com/gp/v/save/${token}`);
+      setSerial(GOOGLE_WALLET.OBJECT_ID(backupSerial));
     };
 
     loadBackup();
-  }, [jubmojis, getClaims]);
+  }, [jubmojis]);
 
-  const isDisabled = !jubmojis || !saveUrl || !updateBackup || !serial;
+  const isDisabled = !jubmojis || !saveUrl || !serial;
 
-  const onAddToWallet = () => {
+  const onAddToWallet = async () => {
     if (isDisabled) return;
 
-    if (updateBackup) {
-      fetch(saveUrl).then((response) => {
-        if (response.status === 200) {
-          setUpdated(true);
-        }
-      });
-    } else if (!updateBackup) {
-      saveBackupState({
-        type: "google",
-        serialNum: serial,
-      });
-      window.location.href = saveUrl;
+    if (updateUrl) {
+      const response = await fetch(updateUrl);
+      if (response.status === 200) {
```

Comment:
What does it mean if the response is not 200? In this case, should we run the saveBackupState flow below?